File: apps/java/src/main/java/chapter12/Experiment.java
package chapter12;

/**
 * 正則化の強さ 1 つ分の実験結果。作ったあとで変えられない。
 *
 * @param alpha 正則化の強さ
 * @param trainScore 訓練データの決定係数
 * @param validationScore 検証データの決定係数
 * @param coefficientAbsSum 係数の絶対値の合計
 */
public record Experiment(
    double alpha, double trainScore, double validationScore, double coefficientAbsSum) {}

File: apps/php/src/Chapter12/Experiment.php
<?php

declare(strict_types=1);

namespace GettingStartedMl\Chapter12;

/** 正則化の強さを 1 つ試した結果。 */
final readonly class Experiment
{
    public function __construct(
        /** 試した正則化の強さ。 */
        public float $alpha,
        /** 訓練データの決定係数。 */
        public float $trainScore,
        /** 検証データの決定係数。 */
        public float $validationScore,
        /** 係数の絶対値の合計。正則化が強いほど小さくなる。 */
        public float $coefficientAbsSum,
    ) {
    }
}
